books/kestrel/utilities/omaps/core.lisp

```diff
@@ -153,7 +156,12 @@
   (defrule alistp-when-mapp
     (implies (mapp x)
              (alistp x))
-    :rule-classes (:rewrite :forward-chaining)))
+    :rule-classes (:rewrite :forward-chaining))
+
+  (defrule consp-when-non-empty-mapp
```

Comment:
This looks like it should perhaps be disabled or rephrased (maybe as a :forward-chaining rule or with a backchain limit), since it may be tried a lot and introduces map reasoning out of nowhere.  Also, the name should probably mention 'car'.